### PR TITLE
improve PVSystem and associated object reprs

### LINKED ISSUE #1027: expand PVSystem repr
The PVSystem repr is

https://github.com/pvlib/pvlib-python/blob/27872b83b0932cc419116f79e442963cced935bb/pvlib/pvsystem.py#L239-L243

The main issue that I have is that the repr doesn't give me enough information about the temperature model settings. It's relatively important because `temperature_model_params` (not printed) may be inferred from `module_type` (not printed) and `racking_model` (printed). So I'd like to add both `temperature_model_params` and `module_type`.

We also don't include `module_parameters`, `inverter_parameters`, and `losses_parameters` in the repr. If I recall correctly, we decided against including these because they can be relatively long. I still think that's reasonable. We could add something like `if len(module_parameters): 'Set. See PVSystem.module_parameters'; else: {}`, but I don't know if that's worth the effort.

### PR DESCRIPTION
**Pull request by wholmgren**
- [x] Closes #1027 
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests added
 - [x] Updates entries to [`docs/sphinx/source/api.rst`](https://github.com/pvlib/pvlib-python/blob/master/docs/sphinx/source/api.rst) for API changes.
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

### CODE REVIEW

**Comment by cwhanse** [comment]
LGTM pending resolution of that odd test failure for py3.5. Thanks @wholmgren

**Comment by wholmgren** [comment]
Ah, dictionary order is not guaranteed until 3.6. I'll specify a dict with a single key.

File: docs/sphinx/source/whatsnew/v0.8.0.rst [hunk 1]
```diff
@@ -44,6 +44,9 @@ Enhancements
   (:pull:`1017`)
 * Add :py:func:`pvlib.inverter.fit_sandia` that fits the Sandia inverter model
   to a set of inverter efficiency curves. (:pull:`1011`)
+* Added *racking_model*, *module_type*, and *temperature_model_parameters* to
```
Comment (kanderso-nrel) [inline docs/sphinx/source/whatsnew/v0.8.0.rst:47]: `*racking_model*` renders as italics on RTD.  Doesn't matter to me, just fyi if those were supposed to be inline literals (double backticks).

Comment (wholmgren) [reply docs/sphinx/source/whatsnew/v0.8.0.rst:47]: I intended for them to be italics but can change them if someone cares. I think too many inline literals make the whats new extremely hard to read.

**Pull request merged**